### ISSUE TITLE
Kill dependency from Google Maps API for geocoding

### ISSUE THREAD

**Issue by kghbln** [JeroenDeDauw/Maps#428]
### Issue

This references https://github.com/SemanticMediaWiki/semantic-mediawiki.org/issues/48

After the change I realize that all maps services provided (openlayers, leaflet and googlemaps) solely rely on googlemaps geocoding for converting addresses into geo coordinates.

Since I have unsuccessfully tried to set up the geocoding API for sm-w.o all Maps examples involving just an address for conversion into coordinates are broken.

Is there an alternative geocoding service to Google and if yes it will be utterly cool to get support for it?

I am pretty saddened by the situation and I would like to present working examples for the Maps extension.

**Comment by JeroenDeDauw** [comment]
Default settings:

```
	// String. The name of the geocoding service to use.
	// Available services: geonames, google, nominatim
	// Some services might require you to provide credentials, see the settings below.
	$GLOBALS['egMapsDefaultGeoService'] = 'geonames';

	// String. GeoNames API user/application name.
	// Obtain an account here: http://www.geonames.org/login
	// Do not forget to activate your account for API usage!
	$GLOBALS['egMapsGeoNamesUser'] = '';
```

If geonames is used but no user is specified, Maps falls back to using the Google geocoder.

**Comment by JeroenDeDauw** [comment]
Improved settings message with https://github.com/JeroenDeDauw/Maps/commit/73f3702b09d989a4dc573e4b157ffab943406969

**Comment by kghbln** [comment]
> Default settings:

Ah, I did not think of this. I just tested with nominatim and geonames and found that nominatim provides better results, i.e. works in spots where geonames does not work. It may however be that I missed how to activate the account property for API usage. 

> Improved settings message with 73f3702

Cool. However I suggest to additional change the default to nominatim since this one does not require additional setup to get results. Moreover this would prevent a googlemaps fallback which is preferred starting June 11 anyways.
